src/init/AppRunner.ts

```diff
@@ -1,41 +1,67 @@
 /* eslint-disable unicorn/no-process-exit */
-
 import type { ReadStream, WriteStream } from 'tty';
-import type { IComponentsManagerBuilderOptions, LogLevel } from 'componentsjs';
+import type { IComponentsManagerBuilderOptions } from 'componentsjs';
 import { ComponentsManager } from 'componentsjs';
 import yargs from 'yargs';
+import type { LogLevel } from '../logging/LogLevel';
+import { LOG_LEVELS } from '../logging/LogLevel';
 import { getLoggerFor } from '../logging/LogUtil';
-import { ensureTrailingSlash, resolveAssetPath, modulePathPlaceholder } from '../util/PathUtil';
+import { createErrorMessage } from '../util/errors/ErrorUtil';
+import { modulePathPlaceholder, resolveAssetPath } from '../util/PathUtil';
 import type { App } from './App';
+import type { CliExtractor } from './cli/CliExtractor';
+import type { VariableResolver } from './variables/VariableResolver';
 
 const defaultConfig = `${modulePathPlaceholder}config/default.json`;
 
-export interface CliParams {
-  loggingLevel: string;
-  port: number;
-  baseUrl?: string;
-  rootFilePath?: string;
-  sparqlEndpoint?: string;
-  showStackTrace?: boolean;
-  podConfigJson?: string;
+const DEFAULT_CLI_RESOLVER = 'urn:solid-server-app-setup:default:CliResolver';
+const DEFAULT_APP = 'urn:solid-server:default:App';
+
+const baseArgs = {
+  config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
+  loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true, choices: LOG_LEVELS },
+  mainModulePath: { type: 'string', alias: 'm', requiresArg: true },
+} as const;
+
+// The components needed by the AppRunner to start up the server
+interface CliResolver {
+  extractor?: CliExtractor;
+  resolver: VariableResolver;
 }
 
 export class AppRunner {
   private readonly logger = getLoggerFor(this);
 
+  /**
+   * Returns an App object, created with the given config, that can start and stop the Solid server.
+   * @param loaderProperties - Components.js loader properties.
+   * @param configFile - Path to the server config file.
+   * @param parameters - Parameters to pass into the VariableResolver.
```

Comment:
That is actually how the `AppRunner` works right now. See my comment here https://github.com/solid/community-server/pull/1065#issuecomment-992277975

I did it like this to be consistent with the solution we already had. Although I don't disagree that it might be interesting to add a function that accepts values for resolved variables. But that would be a new feature.

Reply:
Yeah, I think we unnecessarily have converted every method in this file to the 2-step process, whereas I think we could have still kept some of the original 1-step process and just provided new methods for the 2-step. (There is the reuse issue of the ComponentsManager, but that could be an optional param.)

Up to you.